azure/ConsiderationsForServiceDesign.md

```diff
@@ -141,94 +144,205 @@ cannot collide with a resource path that contains user-specified resource ids.
-If a new operation request is received when there is already a long-running operation in progress for a resource,
-the service should reject the operation if it is inconsistent with one already in progress.
-However, if the new operation is redundant or not inconsistent with the one in progress,
-for example a "reboot" operation on a VM that is in the process of rebooting, then the service should
-accept the request. The status field of the resource should then report the completion status of _both_
-operations.
+The operation is initiated with a POST operation and the operation path ends in `:action`.
 
-Note: The RELO pattern should not be used in cases where the completion status of individual operations
-may be important to users, as opposed to simply learning that an operation of the type they requested
-(e.g. create a resource with a specific name) has successfully completed.
+```text
+POST /<service-or-resource-url>:action 
+Operation-Id: 22 
+ 
+{ 
+   "arg1": 123 
+   "arg2": "abc" 
+} 
+```
 
-### Long-running operations with status monitor
+The response is a `202 Accepted` as described above.
+
+```text
+HTTP/1.1 202 Accepted
+Operation-Location: https://<status-monitor-endpoint>/22
+Retry-After: 5 
+ 
+{
+   "id": "22",
+   "status": "NotStarted"
+}
+```
 
-In the LRO with status monitor pattern, the status and results of the operation are encapsulated into
-a status monitor resource that is distinct from the target resource and specific to the individual
-operation request.  Here's what the status monitor LRO pattern looks like:
+The client will issue a GET to the status monitor to obtain the status and result of the operation.
 
-<!-- markdownlint-disable MD033 -->
-<p align="center">
-  <img src="./statmon.jpg" alt="The status monitor LRO flow"/>
-</p>
-<!-- markdownlint-enable MD033 -->
+```text
+GET https://<status-monitor-endpoint>/22?api-version=2022-05-01
+```
 
-1. The client sends the request to initiate the long-running operation.
-As in the RELO pattern, the initial request could be a PUT, PATCH, POST, or DELETE method.
+When the operation completes successfully, the result (if there is one) will be included in the `result` field of the status monitor.
 
-2. The resource validates the request and initiates the operation processing.
-It sends a response to the client with a `202-Accepted` HTTP status code.
-Included in this response is an `Operation-location` response header with the absolute URL of
-status monitor for this specific operation.
-The response also includes a `Retry-after` header telling the client a minimum time to wait (in seconds)
-before sending a request to the status monitor URL.
+```text
+HTTP/1.1 200 Ok
+ 
+{
+   "id": "22",
+   "status": "Succeeded",
+   "result": { ... }
+}
+```
 
-3. After waiting at least the amount of time specified by the previous response's `Retry-after` header,
-the client issues a GET request to the status monitor URL.
+### Create (PUT) with additional long-running processing
```

Comment:
I might be missing something. But in practice, PUT is often used to either create and update. Is there a different behavior  for updating existing resources with a long-running operation? I feel the guideline can be more prescriptive on it and looks like it should be consistent with create case, i.e. updated content is applied first before LRO is finished.

Reply:
We do NOT consider PUT to create or update; we consider PUT to be create or wholesale replace. We consider PATCH to be create or update. The reason for this is due to Azure's versioning policies. I do have a video describing this [here](https://www.youtube.com/watch?v=PFK6RRTwXCg).

Reply:
Jeff, "update" might be a bit misleading but it is used in many Azure API docs for PUT. One example is "[Create Or Update](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update)" . But my point is to make the guideline specific on the behavior of PUT in the case of replacement. This can be as simple as changing the title here with  "Create Or Replace (PUT)".

Reply:
To update a resource via an LRO, we prescribe the POST-action LRO pattern.

Reply:
I changed the wording a bit to make clear that the "PUT" pattern applies to both "create" and "replace" (since they really aren't separable).